### DISCUSSION TITLE
"Front-end" do futuro

### DISCUSSION THREAD

**Discussion by tarciozemel** [frontendbr/forum#1005] [Discussão]
Deem uma olhadinha neste video/protótipo, por favor:

https://cf-e2.streamablevideo.com/video/mp4/nrzaz.mp4?token=1521908674-dyJc1C9uwXZ%2Bdh0te1tjNLTiuJaxTn2P3pIfHOc2f5E%3D

Quem gosta de um bom "front-end" que se prepare para aprender uma ou duas coisinhas novas para daqui a não muito tempo...

Você acha que o front-end do futuro tem que ser como?

![Image of Yaktocat](https://i.imgur.com/cW8ApPa.png)

**Comment by woliveiras** [comment]
Vídeo bem da hora, por sinal! 

Já existem algumas iniciativas legais para AR no frontend (se tratando de navegador/JavaScript) e isso tende a crescer muito:

- https://aframe.io/
- https://github.com/jeromeetienne/ar.js
- https://www.argonjs.io/

Esse artigo é bem maneiro sobre usar AR.js com Aframe: https://aframe.io/blog/arjs

Não acredito que vai ter uma cobrança tão grande em cima de todas as áreas de negócio, mas algumas pessoas já podem começar a investir tempo nisso se quiserem inovar em seus meios.

Um exemplo legal seria pegar as medidas de um móvel e sua imagem com cor personalizada e poder colocar ele virtualmente em um espaço na minha casa antes de realmente realizar a compra, sabendo se vai caber no espaço e se fica legal no ambiente. Empresas de varejo, por exemplo, se beneficiariam muito disso e trariam uma experiência muito boa para seus clientes.

**Comment by napalmdev** [comment]
caramba, AR é animal d+, e o impacto disso pro cliente é gigante, como vc disse existem casos específicos em que isso se aplica mas se bem aplicado pode alavancar muita venda, foda d+

**Comment by ninetails** [comment]
Inicialmente vejo isto muito mais aplicado a eventos/marketing que a web. E também não sei se aplicaria tanto a desenvolvimento frontend inicialmente e sim a galera de Jogos, a curto/médio prazo, claro.

**Comment by SrVieira** [comment]
Tenho a mesma opinião do @ninetails em relação a aplicação desta tecnologia, mas há uma malha de possibilidades futuras para aplicar-se o AR, mas no atual momento vejo que essa tecnologia seria bem utilizada em divulgação de campanhas de marketing. Exemplo: Coca Cola faz uma campanha com as latinhas que traz prêmios mas para isso o cliente/comprador tem que passar por um gamefication.
